dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -346,73 +346,64 @@ protected void githubWebhookDelete(String repository, String gitReference, Strin
-        GHRateLimit endRateLimit;
-        boolean isSuccessful = false;
-        try {
 
+        boolean isSuccessful = true;
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter messageWriter = new PrintWriter(stringWriter);
+
+        try {
             SourceFile dockstoreYml = gitHubSourceCodeRepo.getDockstoreYml(repository, gitReference);
             // If this method doesn't throw an exception, it's a valid .dockstore.yml with at least one workflow or service.
             // It also converts a .dockstore.yml 1.1 file to a 1.2 object, if necessary.
             final DockstoreYaml12 dockstoreYaml12 = DockstoreYamlHelper.readAsDockstoreYaml12(dockstoreYml.getContent());
-            List<WorkflowAndVersion> updatedWorkflowAndVersions = new ArrayList<>();
-            updatedWorkflowAndVersions.addAll(createServicesAndVersionsFromDockstoreYml(dockstoreYaml12.getService(), repository, gitReference, installationId, user, dockstoreYml));
-            updatedWorkflowAndVersions.addAll(createBioWorkflowsAndVersionsFromDockstoreYml(dockstoreYaml12.getWorkflows(), repository, gitReference, installationId, user, dockstoreYml, false));
-            updatedWorkflowAndVersions.addAll(createBioWorkflowsAndVersionsFromDockstoreYml(dockstoreYaml12.getTools(), repository, gitReference, installationId, user, dockstoreYml, true));
-            LambdaEvent lambdaEvent = createBasicEvent(repository, gitReference, username, LambdaEvent.LambdaEventType.PUSH);
-            addValidationsToLambdaEvent(lambdaEvent, updatedWorkflowAndVersions);
-            lambdaEventDAO.create(lambdaEvent);
-            endRateLimit = gitHubSourceCodeRepo.getGhRateLimitQuietly();
-            isSuccessful = true;
-            gitHubSourceCodeRepo.reportOnGitHubRelease(startRateLimit, endRateLimit, repository, username, gitReference, isSuccessful);
-        } catch (CustomWebApplicationException | ClassCastException | DockstoreYamlHelper.DockstoreYamlException | UnsupportedOperationException ex) {
-            endRateLimit = gitHubSourceCodeRepo.getGhRateLimitQuietly();
-            gitHubSourceCodeRepo.reportOnGitHubRelease(startRateLimit, endRateLimit, repository, username, gitReference, isSuccessful);
-            String errorMessage = ex instanceof CustomWebApplicationException ? ((CustomWebApplicationException)ex).getErrorMessage() : ex.getMessage();
-            String msg = "User " + username + ": Error handling push event for repository " + repository + " and reference " + gitReference + "\n" + errorMessage;
+
+            checkDuplicateNames(dockstoreYaml12.getWorkflows(), dockstoreYaml12.getTools());
+
+            List<Service12> services = dockstoreYaml12.getService() != null ? List.of(dockstoreYaml12.getService()) : List.of();
+            isSuccessful &= createWorkflowsAndVersionsFromDockstoreYml(services, repository, gitReference, installationId, username, dockstoreYml, Service.class, messageWriter);
```

Comment:
This series of `&=` could use a comment. It makes sense that you're using it on purpose to skip the right side methods if anything is unsuccessful but its not too common so could use an explanation.

Reply:
`&=` does not short-circuit:

https://stackoverflow.com/questions/3152208/do-and-short-circuit-in-java

We want to evaluate the right side of those expressions, otherwise we won't read some of the "blocks" of entries.

Will add a comment.